include/pros/detail.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * \file pros/detail.hpp
+ *
+ * Contains PROS internal functions used in port literal functions.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2023 Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License v. 2.0. If a copy of the MPL was not distributed with this
+ * file You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
+
+#include <array>
+#include <charconv>
+#include <cstdint>
+
+namespace pros::detail {
+template <char ...Chars>
+consteval uint8_t parse_port() {
+    const std::array chars{Chars...};
+    const char* str = chars.begin();
+    int base = 10;
+    uint8_t port_num{0};
```

Comment:
i'd move this declaration to right above the `std::from_chars` call because the type of this argument is actually vital to what the function does (continually blown away by how much the `std::from_chars` API sucks!)